enacts/wat_bal/agronomy.py

```diff
@@ -225,7 +294,7 @@ def soil_plant_water_balance(
         planting_date = (peffective[time_dim][-1].drop_vars(time_dim)
             - (planted_since - np.timedelta64(1, "D"))
         )
-    return sm, drainage, et_crop, planting_date
+    return sm, drainage, et_crop, et_crop_red, planting_date
```

Comment:
`sm` is the final result, and all the other arrays are just intermediate quantities that went into calculating it, right? Are you sure we really need  to return them all? If we didn't need to hang on to the intermediate results for dates we've already processed, then we could simplify the code a little.

Reply:
not really. Users would want to look at different things. See this [Maproom](http://iridl.ldeo.columbia.edu/maproom/Water/BAWKP/Agriculture/Bicol_WB_Dry.html#tabs-2) text (Maproom maps might not work very well) to see what folks typically want to do with these.

BUT, some of what is output here can be recomputed from others. So we could output less and then have some other functions to derive again variables of interest. Not sure if it is a gain...

Reply:
> some of what is output here can be recomputed from others. So we could output less and then have some other functions to derive again variables of interest. Not sure if it is a gain...

No, probably not.